### PR TITLE
Add e2e timeout to Dockerhub runs and update Makefile default

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

Earlier e2e test timeout was only applicable for GHCR cron runs, now it's also extended to Dockerhub.
The Makefile is also updated to always use a default value if the environment variable is unset.

## Issue reference

NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by shubham1172** [comment]
MacOS

gotestsum --jsonfile /Users/runner/work/cli/cli/test-e2e-standalone.json --format standard-verbose -- -timeout **20m** -count=1 -tags=e2e ./tests/e2e/standalone/...


Linux

gotestsum --jsonfile /home/runner/work/cli/cli/test-e2e-standalone.json --format standard-verbose -- -timeout **10m** -count=1 -tags=e2e ./tests/e2e/standalone/...

File: Makefile [hunk 1]
```diff
@@ -70,7 +70,11 @@ export GOOS ?= $(TARGET_OS_LOCAL)
 export BINARY_EXT ?= $(BINARY_EXT_LOCAL)
 
 TEST_OUTPUT_FILE ?= test_output.json
-E2E_SH_TEST_TIMEOUT ?= 10m
+
+# Set the default timeout for tests to 10 minutes
```
Comment (pravinpushkar) [inline Makefile:74]: is not this same as `E2E_SH_TEST_TIMEOUT ?= 10m.

Comment (shubham1172) [reply Makefile:74]: Turns out, no. `?=` only sets a variable if it's not already set. In our case, it was set, but to `""`.

**Review by pravinpushkar — APPROVED** [review]

**Review by mukundansundar — COMMENTED** [review]
@shubham1172 one suggestion ... for the self hosted e2e alone keep the timeout at 15 mins .... It is only timeout and that should not affect the actual tests ... 
The point is do we need the complex conditions to set different values based on arch?

**Comment by shubham1172** [comment]
> @shubham1172 one suggestion ... for the self hosted e2e alone keep the timeout at 15 mins .... It is only timeout and that should not affect the actual tests ... The point is do we need the complex conditions to set different values based on arch?

It's better to fail early if something goes wrong, isn't it? The difference is mostly due to podman vs docker in Mac and other runners. We CAN revert this to use a single value, but let's do it in a separate PR since a part of this logic is already merged to `master`, but I am OK with keeping different timeouts too. Let me know your thoughts.

**Review by mukundansundar — APPROVED** [review]

**Comment by mukundansundar** [comment]
> > @shubham1172 one suggestion ... for the self hosted e2e alone keep the timeout at 15 mins .... It is only timeout and that should not affect the actual tests ... The point is do we need the complex conditions to set different values based on arch?
> 
> It's better to fail early if something goes wrong, isn't it? The difference is mostly due to podman vs docker in Mac and other runners. We CAN revert this to use a single value, but let's do it in a separate PR since a part of this logic is already merged to `master`, but I am OK with keeping different timeouts too. Let me know your thoughts.

IMO it can be done as a separate PR.

**Pull request merged**